pkg/labels/oplabels.go

```diff
@@ -201,9 +201,16 @@ func (o *OpLabels) ModifyIdentityLabels(addLabels, delLabels Labels) (changed bo
 // upsertLabel updates or inserts 'label' in 'l', but only if exactly the same label
 // was not already in 'l'. Returns 'true' if a label was added, or an old label was
 // updated, 'false' otherwise.
-func (l Labels) upsertLabel(label Label) bool {
+// The label is only updated if its source matches the provided 'sourceFilter'
+// or in case the provided sourceFilter is 'LabelSourceAny'. The new label must
+// also match the old label 'source' in order for it to be replaced.
```

Comment:
> The new label must also match the old label 'source' in order for it to be replaced.

nit (non-blocking): I was slightly confused why the commit adding this godoc comment did not also add the check (the check is only added in the next commit). Might be worth moving that part of the go doc change into the next commit.